modules/private/class-jetpack-private.php

```diff
@@ -0,0 +1,329 @@
+		add_filter( 'preprocess_comment', array( __CLASS__, 'privatize_blog_comments' ), 0 );
+		add_action( 'blog_privacy_selector', array( __CLASS__, 'privatize_blog_priv_selector' ) );
+		add_filter( 'robots_txt', array( __CLASS__, 'private_robots_txt' ) );
+		add_action( 'wp_head', array( __CLASS__, 'private_no_pinning' ) );
+		add_action( 'admin_init', array( __CLASS__, 'private_blog_prevent_requests' ), 9 );
+		add_action( 'check_ajax_referer', array( __CLASS__, 'private_blog_ajax_nonce_check' ), 9, 2 );
+		add_action( 'rest_pre_dispatch', array( __CLASS__, 'disable_rest_api' ) );
+		add_filter( 'jetpack_get_available_modules', array( __CLASS__, 'private_get_modules' ) );
+		add_action( 'update_option_blog_public', array( __CLASS__, 'private_update_option_blog_public' ) );
+		add_action( 'update_right_now_text', array( __CLASS__, 'add_private_dashboard_glance_items' ) );
+		add_action( 'jetpack_sync_before_send_queue_full_sync', array( __CLASS__, 'remove_privatize_blog_mask_blog_name_filter' ) );
+		add_action( 'jetpack_sync_before_send_queue_sync', array( __CLASS__, 'remove_privatize_blog_mask_blog_name_filter' ) );
+		add_action( 'opml_head', array( __CLASS__, 'hide_opml' ) );
+	}
+
+	/**
+	 * Returns the private site template for private blogs
+	 *
+	 * @param object $wp Current WordPress environment instance (passed by reference).
+	 */
+	public static function privatize_blog( $wp ) {
+		global $pagenow, $wpdb;
+		$this_current_user = wp_get_current_user();
+
+		if ( 'wp-login.php' === $pagenow ) {
+			return;
+		}
+
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			return;
+		}
+
+		// Serve robots.txt for private blogs.
+		if ( is_object( $wp ) && ! empty( $wp->query_vars['robots'] ) ) {
+			return;
+		}
+
+		if ( is_user_logged_in() && ( is_super_admin() || self::is_private_blog_user( $wpdb->blogid, $this_current_user ) ) ) {
+			return;
+		}
+
+		include JETPACK__PLUGIN_DIR . '/modules/private/private.php';
+
+		exit;
+	}
+
+	/**
+	 * Does not check whether the blog is private. Accepts blog and user in various types.
+	 * Returns true for super admins.
+	 *
+	 * @param int   $blog Current WordPress blod id.
+	 * @param Mixed $user Current WordPress user or user_id.
+	 */
+	public static function is_private_blog_user( $blog, $user ) {
+		global $wpdb;
+
+		if ( ! is_object( $user ) ) {
+			$user = new WP_User( $user );
+		}
+
+		if ( ! $user->ID ) {
+			return false;
+		}
+
+		if ( is_numeric( $blog ) ) {
+			$blog_id = intval( $blog );
+		} elseif ( is_object( $blog ) ) {
+			$blog_id = $blog->blog_id;
+		} elseif ( is_string( $blog ) ) {
+			$fields  = array(
+				'domain' => $blog,
+				'path'   => '/',
+			);
+			$blog    = get_blog_details( $fields );
+			$blog_id = $blog->blog_id;
+		} else {
+			$blog_id = $wpdb->blogid;
+		}
+
+		// check if the user has read permissions.
+		$the_user = wp_clone( $user );
+		$the_user->for_site( $blog_id );
+
+		/**
```

Comment:
The hook does not appear elsewhere, so I believe the docblock wouldn't have a repeated use form. That said, I noticed that the docblock content needed updation, so that's done in [c1f7c32](https://github.com/Automattic/jetpack/pull/10333/commits/c1f7c32982523d1f6b0d1a58036b30b39d9b970f)

Reply:
👍 I think when I reviewed it originally, it was used in multiple places. Marking as resolved.